### ISSUE TITLE
ClobBinding and BlobBinding shouldn't delegate sql() generation to internal DefaultStringBinding or DefaultBytesBinding

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14027]
The `ClobBinding` and `BlobBinding` don't work correctly because they emit a nested bind value:

```java
class ClobBinding {
    @Override
    public final void sql(BindingSQLContext<String> ctx) throws SQLException {
        ctx.render().visit(DSL.val(ctx.value(), SQLDataType.CLOB));
    }
}

class BlobBinding {
    @Override
    public final void sql(BindingSQLContext<byte[]> ctx) throws SQLException {
        ctx.render().visit(DSL.val(ctx.value(), SQLDataType.BLOB));
    }
}
```

Instead, the method should a more low level SQL generation:

```java
@Override
public final void sql(BindingSQLContext<String> ctx) throws SQLException {
    if (ctx.render().paramType() == ParamType.INLINED)
        ctx.render().visit(DSL.inline(ctx.convert(converter()).value(), SQLDataType.CLOB));
    else
        ctx.render().sql(ctx.variable());
}
```

The whole part of the `Binding` is to do this work itself, rather than using internals. Otherwise, if a `Param` produces nested `Param` values, then we ignore the parent `Param`, as can be seen here in `DefaultRenderContext`:

```java
        // [#4650] In PostgreSQL, UDTConstants are always inlined as ROW(?, ?)
        //         as the PostgreSQL JDBC driver doesn't support SQLData. This
        //         means that the above internal.accept(this) call has already
        //         collected the bind variable. The same is true if custom data
        //         type bindings use Context.visit(Param), in case of which we
        //         must not collect the current Param
        if (after == before && paramType != INLINED && internal instanceof Param) {
            Param<?> param = (Param<?>) internal;

            if (!param.isInline()) {
                bindValues.add(param);
```

This is due to #4650. It seems that `ClobBinding` and `BlobBinding` never did what they promised?

**Comment by lukaseder** [comment]
Our tests so far have only covered reading lobs from `ResultSet`, not setting them to `PreparedStatement`, which is why this has gone undetected

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14028), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/14029)

**Comment by lukaseder** [comment]
Related:

- https://github.com/jOOQ/jOOQ/issues/18609
